### DISCUSSION TITLE
django-typer

### DISCUSSION THREAD

**Discussion by williln** [django-commons/membership#107] [Check-ins]
Hi @django-commons/django-typer! I wanted to take a moment and check in with you about how the transition to Django Commons has been.

- Have you run into any friction having your project in Django Commons, if so, can you describe them?
- Is there anything we can do to help your team or repo out?
- Do you have any questions or concerns for us?

If you'd like to chat with us privately, our email is [django-commons-admin@googlegroups.com](mailto:django-commons-admin@googlegroups.com). Thanks!

**Comment by bckohan** [comment]
> - Have you run into any friction having your project in Django Commons, if so, can you describe them?

It's been smooth! I tweaked some of the base GA scripts a little. I have one that checks to see if the CoC has updated once a week and if it has it opens a PR to pull in the new CoC. I did this because if it changes I wanted some automatic notification as to what the changes were. I also tweaked the release GA a bit but gave up trying to get it to do what I want it to do which is run my tests first and only publish if everything passes. There seems to be some conditional flow control limitations wrt to how a GA can run other workflows. 

> - Is there anything we can do to help your team or repo out?

Not anything urgent! It'd be nice if the CoC check could be a push (webhook) instead of a pull but ¯\\_(ツ)_/¯

> - Do you have any questions or concerns for us?

I'm self conscious about how long the CI runs on this repo (~20 minutes). There's no way to avoid starting up a bunch of python subprocesses to test this library because of what it is. Every time this happens all the pytest/coverage stuff has to bootstrap which takes awhile so there's not much more efficiency I can wring out of it. I'm worried it might be an unnecessarily large hog on the actions quota. I'm not sure how concerned I should be about this. I'm considering disabling CI on commits and only running it on PRs.

**Reply by tim-schilling** [reply]
>  if the CoC check could be a push (webhook) instead of a pull

That's an interesting idea! I do hope that if we make a change to the CoC there'd be a discussion post here tagging all members. But we haven't run into that use case, so I'm not sure we can promise that.

> I'm worried it might be an unnecessarily large hog on the actions quota

For now, this isn't a problem. Public repos don't have quotas as far as I know. But it's nice to reduce run-times for the environment :green_heart:

**Reply by cunla** [reply]
> if the CoC check could be a push (webhook) instead of a pull

I noticed that GitHub workflow, it is very cool! However, I think it is a bit of overkill, a simple link to the django-commons CoC would suffice.

**Reply by bckohan** [reply]
I thought about just linking, but I'm a pedant and not having the full CoC available when clicking the CoC tab bothered me. And also, I wanted an automagic notification with a nice diff to see when and what any updates were. Not because I don't trust the updates, mostly because I figure if there is an update it was precipitated by something and I'd like the push to figure out what that was.

**Comment by bckohan** [comment]
@cunla @williln @tim-schilling 

Just ran into an issue, but it may be user error. I'm trying to create a fine grained access token for django-typer but it seems I do not have the permissions to do so. Usually the way I do this for a repo owned by an organization is Profile -> Organizations -> django-commons -> settings, but the settings link is not available to me for django-commons. And django-typer is not available to me in the token drop down under my own profile's settings.

Am I missing something?

**Reply by cunla** [reply]
Ni @bckohan, what about if you go to: https://github.com/settings/personal-access-tokens/new

Can you create PAT with access to django-typer there?

**Reply by bckohan** [reply]
no, I no longer own the repo so it doesnt show up in the drop down

<img width="525" alt="Screenshot 2025-01-06 at 3 41 27 PM" src="https://github.com/user-attachments/assets/61a5f8de-c134-45fe-a78c-3bdf1fa457e7" />

**Reply by cunla** [reply]
above you have resource owner... you can't select django-commons?

**Reply by tim-schilling** [reply]
To clarify, further up on the screen there's a resource owner dropdown. It defaults to you. It would need to be django-commons which should then show django commons repo's you're an admin for (which includes django-typer).

**Reply by bckohan** [reply]
!! thankyou! what a terrible UI haha

**Comment by tim-schilling** [comment]
Hi @django-commons/django-typer, we're circling back around :dizzy: We want to ask [Lacey's original questions](https://github.com/orgs/django-commons/discussions/107#discussion-7490228) to see if there's any update. We're also looking into how can we improve the new contributor experience with the overall goal of helping find regular contributor and more maintainers for projects.
- [Meet and greets](https://github.com/orgs/django-commons/discussions/136)
- [Project introduction videos](https://github.com/orgs/django-commons/discussions/150)

If any of that sounds interesting to you, feel free to engage on the discussions or here.

**Comment by tim-schilling** [comment]
I approved the token requests that grants access to django-typer and django-enum repos you requested @bckohan. No response needed. This is more for traceability
cc: @django-commons/admins

**Reply by bckohan** [reply]
@tim-schilling @django-commons/admins  What token requests? My GitHub account was compromised as part of the nx supply chain attack that hit vscode via the NX Console app. Was this request made between Tuesday and Thursday?

If these were the OSSF Scorecard tokens - that was me.

**Reply by Stormheg** [reply]
@bckohan the request was indeed for the OSSF Scorecard. Looks like you made the request around Friday.

<img width="967" height="605" alt="Screenshot 2025-08-31 at 10 55 38 AM" src="https://github.com/user-attachments/assets/f9a9051b-b707-4fcf-af0f-9529e710edf2" />

**Reply by tim-schilling** [reply]
Sorry, I revoked the token out of an abundance of caution. There were no other actions taken that would indicate it wasn't something valid. If you'd like that token recreated, please re-request it again.

Audit log from the time it was granted:
<img width="945" height="551" alt="Screenshot from 2025-08-31 05-40-57" src="https://github.com/user-attachments/assets/d3a88bc1-c107-40d7-a4fd-35704ded7428" />

**Reply by bckohan** [reply]
Thanks! I will re-request. I should have notified the django-commons admin group last week, but I was so busy putting out other fires I didn't get to it. There was more than a single vector, but if you happened to be one of the 2 million vscode users that had NX Console installed last week and restarted within the 5 hour exploit window, at the very least your github CLI token was leaked (if you were lucky it was just that).

**Comment by williln** [comment]
Hi @django-commons/django-typer -- this is just our semi-annual checkin. I see you had some PAT token issues a couple weeks back but it seems those are resolved? Let us know if they are not resolved or if there is anything else you need! Otherwise, happy coding 🌳

**Reply by bckohan** [reply]
@williln Yep everything is fine!

**Comment by bckohan** [comment]
@cunla @williln @tim-schilling I invited the django commons email to be a maintainer of the docs on rtd.org - the invite expires in a week, just FYI.

**Reply by tim-schilling** [reply]
Thank you for the heads up. I've accepted it.

**Comment by tim-schilling** [comment]
Hi @django-commons/django-typer this is just our semi-annual checkin. Let us know if there is anything Django Commons can do to help, or if there is anything else on your mind! Otherwise, happy coding 🥳

As an additional reminder, though not at all a suggestion/requirement. It may be worth looking through who has regularly contributed to your project to see if you'd like to elevate them to be a committer or a maintainer. I mention it because it's easy to forget to review the repositories community and helping people grow. Similarly, if you or anyone else wants to step back, that's fine too! If we can help coordinate, please let us know.

**Reply by tim-schilling** [reply]
Whoops, forgot some things. There are two other things:

1. Can the @django-commons/django-typer-admins please share your emails with us? You can email us at [django-commons-admin@googlegroups.com](mailto:django-commons-admin@googlegroups.com) to avoid publishing publicly here.
2. We are working on a new policy around project abandonment. As a community organization, we need to have clear guidelines on when the community should pass the project onto someone else or signaling that the project is in a period of low activity. We'll share this new policy with you when we have it ready for comment. This will include a timeline for adoption. If you have questions, please let us know!

**Reply by bckohan** [reply]
All is well!
